test/e2e/redundancy/gateway_failover.go

```diff
@@ -41,12 +61,24 @@ func testEnginePodRestartScenario(f *subFramework.Framework) {
 	enginePod := f.AwaitSubmarinerEnginePod(framework.ClusterA)
 	By(fmt.Sprintf("Found submariner engine pod %q on %q", enginePod.Name, clusterAName))
 
+	By(fmt.Sprintf("Ensuring that the gateway reports as active %q", clusterAName))
+	activeGateways := f.GetGatewaysWithHAStatus(subv1.HAStatusActive)
+	Expect(activeGateways).To(HaveLen(1))
+
+	gw := activeGateways[0]
+	By(fmt.Sprintf("Ensuring that the gateway %q is reporting connections", gw.Name))
+	Expect(gw.Status.Connections).NotTo(BeEmpty())
+	Expect(gw.Status.Connections[0].Status).To(Equal(subv1.Connected))
+
 	By(fmt.Sprintf("Deleting submariner engine pod %q", enginePod.Name))
 	f.DeletePod(framework.ClusterA, enginePod.Name, framework.TestContext.SubmarinerNamespace)
 
 	newEnginePod := f.AwaitSubmarinerEnginePod(framework.ClusterA)
 	By(fmt.Sprintf("Found new submariner engine pod %q", newEnginePod.Name))
 
+	By(fmt.Sprintf("Waiting for the gateway to be up and connected %q", newEnginePod.Name))
+	f.AwaitGatewayFullyConnected(framework.ClusterA, gw.Name)
```

Comment:
This could report fully connected as the last state from the previous engine pod.  Perhaps we should move this after the connectivity tests below - that way we know the gateway is re-connected and should be reporting the new state.

Reply:
Oh, that is true, I was actually wanting to make sure we had connections before starting the dataplane tests.

This will work better/faster when we I implement the cleanup of state at pod exit, I didn't dig into that, but I will do right after 0.3.0-rc1
https://github.com/submariner-io/submariner/issues/518

Reply:
yeah I was thinking that too - delete on engine shutdown.

Reply:
I added a DeleteGateway function for now, right after the pod deletion, which will ensure that at least the gateway status is re-created, and that it eventually moves to "fully connected"

When I do #518 then we remove this, and probably do a different check (I guess an informer to make sure it's deleted and re-created).